src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -36,9 +32,12 @@ class DiscoverQuerySerializer(serializers.Serializer):
         required=True,
         allow_null=False,
     )
-    start = serializers.DateTimeField(required=False)
-    end = serializers.DateTimeField(required=False)
+    start = serializers.CharField(required=False)
+    end = serializers.CharField(required=False)
```

Comment:
i made this change because `get_date_range_from_params` handles conversion from string to datetime and by allowing this to be handled by that function, we'll also get some additional validation checking for whether start is before end

Reply:
I think you'll need `allow_none=True` as well

Reply:
mm not totally sure what you mean. do you mean that if you don't pass `allow_null`, drf will include that field with a value of an empty string for non-required fields? i'm not seeing that behavior and am pretty sure a bunch of the tests would be failing if that were the case (like every one that passes start and end but not range)

Reply:
Yes, a value of `null` will be cast to an empty string if you don't specify `allow_null` which I believe is called `allow_none` in our version of django rest framework. Since you're comparing these values against `None` in the validation code on lines 92 and 109, wouldn't we want these to actually take the value of `None` instead of `""`?

Reply:
are you saying if the request specifically includes `"range": null` (rather than omitting the key entirely?)

Reply:
Yes, discover always sends queries of the same shape even when the key is null. If the key is omitted I think it just won't be in the dictionary so the checks e.g. `data.get('someinvalidkey')` will be None. I think we should make sure we avoid coercing to a string and treat both cases the same because it could cause subtly weird behaviour if it's being sent as null.

Reply:
ok good to know. would be nice for tests to be reflective of how data is actually sent, will update them

Reply:
done

Reply:
thanks